pkg/cmd/dotgit/pull.go

```diff
@@ -14,10 +14,12 @@ import (
 func newCmdPull() *cobra.Command {
 	options := &struct {
 		dir          string
+		checkout     string
 		restore      bool
 		useNativeGit bool
 	}{
 		dir:          filepath.Join(util.CacheDir(), "dotgit"),
+		checkout:     "",
```

Comment:
For extracted repos, it's not deterministic which branch (main or nightly) is HEAD.  It can be confusing for who are not used to be this repo and this command.
If "git switch" cost is light, how about specifying "main" for checkout's default value?